# license: modifies Beetle WonderSwan (https://github.com/libretro/beetle-wswan-libretro), GPL-2.0 (COPYING); this patch is distributed under the same license.
diff --git a/Makefile b/Makefile
index 878e147..54c7129 100644
--- a/Makefile
+++ b/Makefile
@@ -196,6 +196,22 @@ else ifeq ($(platform), ps2)
    STATIC_LINKING = 1
    OLD_GCC = 1
 
+# TRIMUI BRICK / SMART PRO
+else ifeq ($(platform), tg5040)
+   TARGET := $(TARGET_NAME)_libretro.so
+   CC = $(CROSS_COMPILE)gcc
+   CXX = $(CROSS_COMPILE)g++
+   AR = $(CROSS_COMPILE)ar
+   fpic := -fPIC
+   SHARED := -shared -Wl,--no-undefined -Wl,--version-script=link.T
+   LDFLAGS += -lrt
+   FLAGS += -fomit-frame-pointer -ffast-math -flto
+   FLAGS += -mtune=cortex-a53 -mcpu=cortex-a53 -march=armv8-a
+   LDFLAGS += -flto
+   ifneq ($(DEBUG),1)
+      LDFLAGS += -s
+   endif
+
 # PSP
 else ifeq ($(platform), psp1)
    TARGET := $(TARGET_NAME)_libretro_$(platform).a
